src/main/java/org/folio/rest/service/StreamService.java

```diff
@@ -66,16 +68,25 @@ public String enhanceFlux(String firstFluxId, Flux<String> secondFlux, String co
     while(firstString != null || secondString != null) {
       JsonNode firstObject = mapper.readTree(firstString);
       JsonNode secondObject = secondString != null ? mapper.readTree(secondString) : null;
-      JsonNode enhancementNode = secondObject != null ? secondObject.get(comparisonProperty) : null;
-      if (comparator.compare(firstString, secondString) == 0) {
+      JsonNode enhancementNode = secondObject != null ? secondObject.get(enhancementProperty) : null;
+      boolean matched = true;
+      for (Entry<String, String> entry : comparisonMap.entrySet()) {
+        Comparator<String> comparator = nullsLast(new PropertyComparator(entry.getKey(), entry.getValue()));
+        if (comparator.compare(firstString, secondString) < 0) {
+          result = result.concatWith(Flux.just(mapper.writeValueAsString(firstObject)));
+          firstString = firstIter.hasNext() ? firstIter.next() : null;
+          matched = false;
+          break;
+        } else if (comparator.compare(firstString, secondString) > 0) {
+          secondString = secondIter.hasNext() ? secondIter.next() : null;
+          matched = false;
+          break;
+        }
+      };
+      if (matched) {
```

Comment:
As match is declared true in beginning of while loop, this will be called for each key value pair in the map when they equal. If id of stream one matches id of stream two it will enhance stream one tuple and advance both even if the schema does not match. This check probably should be after the for loop of the comparison map.